### PR TITLE
WIP: Nested steps

### PR DESCRIPTION
**Pull request by DavertMik**
Added basic nested step suport

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,164 +23,341 @@ const supportedHelpers = [
...
+module.exports = (config) => {
+  config = Object.assign(defaultConfig, config);
 
-    return rpClient.startLaunch({
-      name: this.config.launchName || suiteTitle,
-      description: this.config.launchDescription || '',
-      attributes: this.config.launchAttributes || [],
-      rerun: this.config.rerun || false,
-      rerunOf: this.config.rerunOf || undefined,
+  console.log('Staaarting!!!11');
```
Comment (kobenguyent) [inline index.js]: why don't we use `output` of codeceptjs here? :)

Comment (DavertMik) [reply index.js]: This is a very dump debug :)

File: index.js [hunk 2]
```diff
@@ -15,164 +23,341 @@ const supportedHelpers = [
...
+  let rpClient;
+  let logFile;
+  let suiteStatus = 'PASSED';
+  let launchStatus = 'PASSED';
+  let currentMetaSteps = [];
+
+
+  event.dispatcher.on(event.all.before, () => {
+    launchObj = startLaunch();
+    console.log(`${launchObj.tempId}: The launchId is started.`);
```
Comment (kobenguyent) [inline index.js]: maybe we can try to use the `debug: true` of RP, then it will print out all those things 😸

Comment (DavertMik) [reply index.js]: Yeah, was thinking about it as well
Also about using `debug` function
Once I make it work I will clean it up

File: README.md [hunk 3]
```diff
@@ -1,41 +1,44 @@
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/6e6495428bbd41f0807e4239c42403eb)](https://www.codacy.com/manual/PeterNgTr/codeceptjs-rphelper?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=PeterNgTr/codeceptjs-rphelper&amp;utm_campaign=Badge_Grade) [![npm version](https://badge.fury.io/js/codeceptjs-rphelper.svg)](https://badge.fury.io/js/codeceptjs-rphelper) [![Greenkeeper badge](https://badges.greenkeeper.io/PeterNgTr/codeceptjs-rphelper.svg)](https://greenkeeper.io/)
 
 # codeceptjs-rphelper
+
 This helpes you integrate the test results of CodeceptJS with ReportPortal
 
 codeceptjs-rphelper is a [CodeceptJS](https://codecept.io/) helper which can publish tests results on [ReportPortal](https://reportportal.io/) after execution.
```
Comment (kobenguyent) [inline README.md]: Probably we should update wordings here. Not to mention helper.

File: README.md [hunk 4]
```diff
@@ -1,41 +1,44 @@
...
 
 ## Installation
+
 ```sh
 npm i codeceptjs-rphelper --save
```
Comment (kobenguyent) [inline README.md]: update the package name as well.

**Comment by DavertMik** [comment]
I updated this PR to include error & screenshots and attach them correctly to the latest failed step.

Issues solved:
* `step.failed` fired multiple times on retry
* nested steps hierarchy was not correctly reported
* browser logs are not available from other helpers than Puppeteer

**Review by kobenguyent — APPROVED** [review]

**Pull request merged**